examples/pyomobook/test_book_examples.py

```diff
@@ -0,0 +1,384 @@
+    
+    # scripts_ch'
+    'test_scripts_ch_warehouse_cuts': ['matplotlib'],
+    
+    # performance_ch'
+    'test_performance_ch_wl': ['numpy','matplotlib'],
+}
+
+
+#
+# Initialize the availability data
+#
+solvers_used = set(sum(list(solver_dependencies.values()), []))
+available_solvers = check_available_solvers(*solvers_used)
+solver_available = {solver_:solver_ in available_solvers for solver_ in solvers_used}
+
+package_available = {}
+package_modules = {}        
+packages_used = set(sum(list(package_dependencies.values()), []))
+for package_ in packages_used:
+    pack, pack_avail = attempt_import(package_)
+    package_available[package_] = pack_avail
+    package_modules[package_] = pack
+
+
+def check_skip(name):
+    """
+    Return a boolean if the test should be skipped
+    """
+
+    if name in solver_dependencies:
+        solvers_ = solver_dependencies[name]
+        if not all([solver_available[i] for i in solvers_]):
+            # Skip the test because a solver is not available
+            _missing = []
+            for i in solvers_:
+                if not solver_available[i]:
+                    _missing.append(i)
+            return "Solver%s %s %s not available" % (
+                's' if len(_missing) > 1 else '',
+                ", ".join(_missing),
+                'are' if len(_missing) > 1 else 'is',)
+
+    if name in package_dependencies:
+        packages_ = package_dependencies[name]
+        if not all([package_available[i] for i in packages_]):
+            # Skip the test because a package is not available
+            _missing = []
+            for i in packages_:
+                if not package_available[i]:
+                    _missing.append(i)
+            return "Package%s %s %s not available" % (
+                's' if len(_missing) > 1 else '',
+                ", ".join(_missing),
+                'are' if len(_missing) > 1 else 'is',)
+
+        # This is a hack, xlrd dropped support for .xlsx files in 2.0.1 which
+        # causes problems with older versions of Pandas<=1.1.5 so skipping
+        # tests requiring both these packages when incompatible versions are found
+        if 'pandas' in package_dependencies[name] and 'xlrd' in package_dependencies[name]:
+            if check_min_version(package_modules['xlrd'], '2.0.1') and \
+               not check_min_version(package_modules['pandas'], '1.1.6'):
+                return "Incompatible versions of xlrd and pandas"
+
+    return False
+
+def filter(line):
+    """
+    Ignore certain text when comparing output with baseline
+    """
+    for field in ( '[',
+                   'password:',
+                   'http:',
+                   'Job ',
+                   'Importing module',
+                   'Function',
+                   'File', 
+                   'Restricted license',
+                   'Using license',
+                   'Matplotlib',
+                   '    ^'):
+        if line.startswith(field):
```

Comment:
Is it possible to combine the `field` lists into one statement that is something like:
```
for field in [<combined list>]:
    if line.startswith(field) or field in line:
        return True
return False
```

Reply:
Yeah, probably. My only concern would be reduced performance as these tests are already pretty slow to run. I'll do some quick profiling to see if the filter is where a significant amount of time is being spent and to compare your suggested to change with the current implementation.

Reply:
Combining the field lists into one statement is a little slower than the current implementation so I'm going to leave it.